ads/contentad.js

```diff
@@ -33,10 +33,12 @@ export function contentad(global, data) {
   window.document.body.appendChild(cadDiv);
 
   /* Capture or pass URL */
+  const h = window.context.location.host;
+  const od = (data.url) ? data.url : window.atob(data.d) ;
   let adUrl = window.context.location.href;
-  if (data.url) {
-    adUrl = adUrl.replace(window.context.location.host, data.url);
-  }
+  const myreg = new RegExp(':\/\/.*?(?=([a-z0-9\-]+\.?)?' + od + ')', 'i');
```

Comment:
You can test the following regex with the domain variable replaced with "nonblocking.io" using the below test site and the below test values. The intent is to remove the CDN domain and any sub-directories in order to simulate the source url for the article containing the Content.ad amp-ad iframe.

http://regexr.com/

-- Google CDN & path --
/:\/\/.*?(?=([a-z0-9\-]+\.?)?nonblocking.io)/i

:::test cases:::
http://www.google.com/c/ot-her.nonblocking.io/examples/ads.amp.max.html#contentad?test=true
http://cdn.google.com/amp/hyphen-ated.nonblocking.io/
http://www.ampproject.org/v/ne-w.nonblocking.io
http://new.nonblocking.io/examples/ads.amp.max.html#contentad?test=true
http://new.nonblocking.io/
http://new.nonblocking.io
http://www.google.com/c/nonblocking.io/examples/ads.amp.max.html#contentad?test=true
http://cdn.google.com/amp/nonblocking.io/
http://www.ampproject.org/v/nonblocking.io
http://nonblocking.io/examples/ads.amp.max.html#contentad?test=true
http://nonblocking.io/
http://nonblocking.io
http://www.google.com/v/wrong-nonblocking.io/examples/ads.amp.max.html#contentad?test=true
http://www.google.com/v/wrong-nonblocking.io-com/
http://www.google.com/v/wrongnonblocking.io
http://wrongnonblocking.iom/examples/ads.amp.max.html#contentad?test=true
http://wrongnonblocking.io/
http://wrongnonblocking.io
http://wrongnonblocking.io-com/
http://wrongnonblocking.io-com